src/ast/semantic_analyser.cpp

```diff
@@ -2007,6 +2007,12 @@ void SemanticAnalyser::visit(Cast &cast)
 {
   cast.expr->accept(*this);
 
+  if (cast.expr->type.IsRecordTy())
+  {
+    LOG(ERROR, cast.loc, err_)
+        << "Cannot cast struct type \"" << cast.expr->type << "\"";
```

Comment:
would be nice to include the target type as well, but that means shuffling a bit

Reply:
right, that'll also break some existing tests but we should do that nevertheless